io-engine/src/grpc/v1/pool.rs

```diff
@@ -1,38 +1,36 @@
 use crate::{
     core::Share,
-    grpc::{rpc_submit, GrpcClientContext, GrpcResult, Serializer},
+    grpc::{rpc_submit, GrpcClientContext, GrpcResult, RWSerializer, ReadLock},
     lvs::{Error as LvsError, Lvs},
     pool_backend::{PoolArgs, PoolBackend},
 };
+use ::function_name::named;
 use futures::FutureExt;
+use io_engine_api::v1::pool::*;
 use nix::errno::Errno;
-use std::{convert::TryFrom, fmt::Debug};
+use std::{convert::TryFrom, fmt::Debug, panic::AssertUnwindSafe};
 use tonic::{Request, Response, Status};
 
-use io_engine_api::v1::pool::*;
-
 #[derive(Debug)]
 struct UnixStream(tokio::net::UnixStream);
 
-use ::function_name::named;
-use std::panic::AssertUnwindSafe;
-
 /// RPC service for mayastor pool operations
-#[derive(Debug)]
+#[derive(Debug, Clone)]
 #[allow(dead_code)]
 pub struct PoolService {
     name: String,
-    client_context: tokio::sync::Mutex<Option<GrpcClientContext>>,
+    pub client_context:
```

Comment:
don't need to expose this

Reply:
WHen doing Stats Reset, We take read lock on PoolService ReplicaService from StatService .